js/src/interface.js

```diff
@@ -274,6 +266,29 @@ let app = new Vue({
     }
   },
   methods: {
+    initDataSourceProvider(currentDataSourceId) {
+      let dataSourceData = {
+        dataSourceTitle: settings.dataSourceLabel ? settings.dataSourceLabel : 'Select data source',
+        dataSourceId: currentDataSourceId,
+        appId: Fliplet.Env.get('appId'),
+        default: {
```

Comment:
```suggestion
        default: _.extend({
          name: `Data for ${Fliplet.Env.get('appName')}`,
          entries: [],
          columns: []
        }, settings.default),
```

Instead of providing `settings.defaultDataSourceName`, we can require the caller component to provider a `settings.default` object, e.g.

```
  default: { name: `User data for ${Fliplet.Env.get('appName')}` }
```

This way, `entries` and `columns` can be set in one go as well if necessary.